src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,43 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final int PLUS_ONE_DAY = 1;
+    private static final int DATE_IN_DATA = 0;
+    private static final int NAME_IN_DATA = 1;
+    private static final int HOURS_IN_DATA = 2;
+    private static final int HOURLY_INCOME_IN_DATA = 3;
+    private static final DateTimeFormatter DATE_TIME_FORMATTER
+            = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder result = new StringBuilder("Report for period ")
+                .append(dateFrom)
+                .append(" - ")
+                .append(dateTo);
+
+        for (String name : names) {
+            int salary = 0;
+            for (String date : data) {
```

Comment:
Please, find another name for variable "String date".
"String date" is not a good name here because it contains not only date.
data - данные
date - дата